lib/zlib.js

```diff
@@ -274,7 +274,7 @@ function ZlibBase(opts, mode, handle, { flush, finishFlush, fullFlush }) {
   this._defaultFlushFlag = flush;
   this._finishFlushFlag = finishFlush;
   this._defaultFullFlushFlag = fullFlush;
-  this.once('end', _close.bind(null, this));
+  this.once('end', this.close);
```

Comment:
Instead of doing this, just remove the event listener and enable `autoDestroy: true`